File: src/assignment/_01To10/P10_RearrangeGivenArray.java
package assignment._01To10;

import java.util.Scanner;

import static java.lang.System.out;

/*
 * program to rearrange the array in such a way that the first element is first maximum
 * and second element is first minimum
 */

// class declared as package private and final to prevent inheritance
final class P10_RearrangeGivenArray {
    // initializing instance for scanner class as constant
    private static final Scanner INPUT = new Scanner(System.in);
    private int arrayCount = 0;

    // method to get integer input from the user
    private static int getIntegerFromUser() throws Exception{
        if(!INPUT.hasNextInt()){
            throw new Exception("Invalid input...");
        }else{
            return INPUT.nextInt();
        }
    }

    // method to get array from the user
    private int[] getArrayFromUser() throws Exception{
        arrayCount++;
        out.print("Enter the number of elements in array"+arrayCount+" : ");
        int arraySize = getIntegerFromUser();
        int [] temp = new int[arraySize];

        out.println("Enter the elements one by one : ");
        // getting array element
        for(int i=0; i<arraySize; i++){
            temp[i] = getIntegerFromUser();
        }
        return temp;
    }

    // method to rearrange the given array
    private int[] rearrangeArray(int[] array){
        int[] newArray = new int[array.length];
        int max = array.length-1, min=0;
        int i =0;
        while(min<=max && i < array.length){
            if(i%2 == 0)
                newArray[i] = array[max--];
            else
                newArray[i] = array[min++];
            i++;
        }

        return newArray;
    }

    // method to print the given array
    private void printThisArray(int[] array){
        out.print("{");
        int count = 0;
        for(int number: array){
            out.print(++count < array.length ? number + "," : number);
        }
        out.print("}");
    }

    public static void main(String [] args) throws Exception{
        P10_RearrangeGivenArray program10 = new P10_RearrangeGivenArray();
        out.println("\nTool to rearrange the number the array ");
        try{
            int[] array = program10.getArrayFromUser();
            array = program10.rearrangeArray(array);
            program10.printThisArray(array);
        }catch( Exception e){
            out.println(e.getMessage());
        }
    }

}

File: src/kotlin_assignment/_01To10/P10_RearrangeGivenArray.kt
package kotlin_assignment._01To10;

/*
 * program to rearrange the array in such a way that the first element is first maximum
 * and second element is first minimum
 */

// method to swap numbers

// method to sort array
fun sortArray(arr:Array<Int>):Array<Int>{
    var temp:Int;
    for(i in arr.indices){
        for ( j in i+1 until arr.size){
            if(arr[i] > arr[j]){
                temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
            }
        }
    }
    return arr;
}

// method to print the array in the given pattern
fun printPattern(arr1:Array<Int>): Array<out Any> {
    var arr1 = sortArray(arr1);
    var arr2:ArrayList<Int> = ArrayList();
    var count:Int=0;
    for(i in arr1.indices){
        if(count > arr1.size/2)
            break;
        if(i%2 == 0) {
            arr2.add(arr1[arr1.size-count-1]);
        }else{
            arr2.add(arr1[count]);
            count++;
        }
    }
    return arr2.toArray();
}

fun main(){
    val arr1:Array<Int> = arrayOf(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11);
    print("Given array in the given pattern : ")
    for(number in printPattern(arr1)){
        print("$number ");
    }
}